### ISSUE TITLE
Bitget) how to make Trigger order with Stop Loss?

### ISSUE THREAD

**Issue by PUVl** [ccxt/ccxt#21487]
### Operating System

Window 10

### Programming Languages

Python

### CCXT Version

Latest

### Description

I want to know how to make order with stop loss.
(market order and trigger order)

++ market order with stop loss is not working and trigger order with stop loss make error(Error: ExchangeError, bitget createOrder() params can only contain one of triggerPrice, stopLossPrice, takeProfitPrice, trailingPercent)

### Code

                type_1 = 'market'
                side_1 = 'buy'
                size_1 = amount_1

                order_1 = bitget.create_order(symbol, type_1, side_1, size_1)

```
                  type_2 = 'market'
                side_2 = 'sell'
                size_2 = amount_2
                price_2 = order_price_1
                params_2 = {
                    'triggerPrice': open_price,
                    'stopLoss': {
                        'triggerPrice': stop_price,
                    }
                }

                order_2 = bitget.create_order(symbol, type_2, side_2, size_2, price_2, params_2)

```

**Comment by Dan-krm** [comment]
Hi @PUVl,

In your first example to create a market order with an 'attached' stop loss, this worked for me:
```
params = {
    'stopLoss': {
        'triggerPrice': 59000,
    }
}

order_1 = bitget.create_order('BTC/USDT:USDT', 'market', 'buy', 0.001, None, params)
```

In your second example, you need to set the `price` in the stopLoss object, it's required on bitget for trigger orders with an attached stop loss. This worked for me:
```
params = {
    'triggerPrice': 60000,
    'stopLoss': {
        'triggerPrice': 59000,
        'price': 58000,
    }
}

order_2 = bitget.create_order('BTC/USDT:USDT', 'market', 'buy', 0.001, None, params)
```

**Comment by Alkem314** [comment]
Hi,

I know this subject is already closed, but I am encountering a similar issue. However, my stop loss is attached to a trigger order, and I want the stop loss to execute at market. The option is configurable directly on the platform, but with CCXT I am required to indicate a price as in your example, and that makes my stop loss act like a limit order. How can I ensure my stop loss will be executed even if the price goes below the limit?

**Comment by Dan-krm** [comment]
Hi @Alkem314 Looking at this endpoint I'm not sure if this functionality is available through their API yet:
https://www.bitget.com/api-doc/contract/plan/Place-Plan-Order

You could try setting the stopLossTriggerType exchange specific param to "fill_price" instead of the defaulted "mark_price" and see if that helps `'{"stopLoss":{"type":"fill_price"}}'`

**Comment by Alkem314** [comment]
hi @Dan-krm Thank you for taking the time to respond to me. 
It doesn't work for me because if I don't specify a price, I encounter an error due to my main order being a trigger order. If I do specify a price, my stop-loss order appears as a limit order regardless of the type. However, I solved this by simulating a market order with a very high or very low price, different from the trigger price. The system tries to execute at the best price available, effectively acting like a market stop-loss order. I have a few more tests to conduct, and while I believe this is not the optimal solution, it works for now. I base my price on a variation of my trigger price because the system doesn't accept absolute values like 0, but this workaround functions.

Now, I'm facing another issue and perhaps you can help. I'm attempting to track my order via sockets (watch_orders()) from its creation until the push of the closing message of the pending stop-loss for the main order.

Here's my problem explained clearly: When I create my main order, I receive an initial message with the clientOid and orderId:
{
    "clientOid": "ADA_USDT_USDT_BTFLY_slice_up_1_1",
    "orderId": "1186545136920240128"
}
When this main order is triggered, I receive subsequent updates, typically in 2 or 3 stages (open / partially filled / closed):
{
    "clientOid": "1186545136920240128",
    "orderId": "1186545557023895553",
    "status": "live", # can be "partially_filled" or "closed"
    "amount": 20.0,
    "filled": 0.0,
    "remaining": 20.0,
    "timestamp": 1718620426424,
    "datetime": "2024-06-17T10:33:46.424Z"
}
As you can see, these updates allow me to easily track my main order using the clientOid, which remains consistent throughout the lifecycle of the order. However, after my main order is closed, the system automatically creates my stop-loss order without notification. When my stop-loss is triggered, I receive another series of updates:
{
    "clientOid": "1186545557663506433",
    "orderId": "1186575243359694853",
    "status": "live",
    "amount": 20.0,
    "filled": 0.0,
    "remaining": 20.0,
    "timestamp": 1718627504196,
    "datetime": "2024-06-17T12:31:44.196Z"
}
To summarize my issue: How can I effectively track the execution of my main order until the triggering of its pending stop-loss? I've already tried different planType parameters for the socket and i tried watch_my_trades but still face the same problem. Using identifiers like symbol isn't feasible due to multiple stop-loss orders for the same symbol (I need to reopen the main order when its stop-loss is triggered, and I have several small orders for a single symbol). 
If you have any ideas that could help, please let me know. And if I need to refine my question or open a new topic, don't hesitate to tell me. Thank you in advance
